pyvista/utilities/geometric_objects.py

```diff
@@ -783,6 +786,10 @@ def Polygon(center=(0.0, 0.0, 0.0), radius=1, normal=(0, 0, 1), n_sides=6):
 
     """
     src = _vtk.vtkRegularPolygonSource()
+    if generate_polygon:
+        src.GeneratePolygonOn()
+    else:
+        src.GeneratePolygonOff()
```

Comment:
Knowing VTK there's probably a method we can call as `src.SetGeneratePolygon(generate_polygon)`.

Reply:
It is worth [trying](https://github.com/pyvista/pyvista/pull/2767/files#r894998106).